### PR TITLE
[MINOR]:Do not introduce unnecessary repartition when row count is 1.

### PR DESCRIPTION
**Pull request by mustafasrepo**
## Which issue does this PR close?

Closes #.

## Rationale for this change
As observed in [discussion](https://github.com/apache/arrow-datafusion/pull/7793#discussion_r1358796170) bu @alamb. Currently we add RoundRobin repartition when we know that input row number is 1(repartition is not helpful). This PR fixes this problem.

### CODE REVIEW

**Review by ozankabak — APPROVED** [review]
Thanks for the quick fix, did a review and it LGTM.

File: datafusion/core/src/physical_optimizer/enforce_distribution.rs [hunk 1]
```diff
@@ -1289,9 +1287,16 @@ fn ensure_distribution(
             maintains,
             child_idx,
         )| {
+            // Don't need to apply when the returned row count is not greater than 1:
+            let stats = child.statistics();
+            let repartition_beneficial_stats = if stats.is_exact {
+                stats.num_rows.map(|num_rows| num_rows > 1).unwrap_or(true)
```
[the remark this answers is not in the document]
Comment (mustafasrepo) [reply datafusion/core/src/physical_optimizer/enforce_distribution.rs:1293]: Makes sense, I will change accordingly.

[the remark this answers is not in the document]
Comment (mustafasrepo) [reply datafusion/core/src/physical_optimizer/enforce_distribution.rs]: @Dandandan  I updated check as you suggested. Some of the existing tests changes with this change. I think, changes are for the better. However, I would appreciate If you can double check them.

File: datafusion/physical-plan/src/aggregates/mod.rs [hunk 2]
```diff
@@ -1021,12 +1021,21 @@ impl ExecutionPlan for AggregateExec {
...
+            _ => {
+                let input_stats = self.input.statistics();
+                // Input statistics is exact and number of rows not greater than 1:
+                let is_exact = input_stats.is_exact
+                    && (input_stats
+                        .num_rows
+                        .map(|num_rows| num_rows <= 1)
+                        .unwrap_or(false));
+                Statistics {
+                    // the output row count is surely not larger than its input row count
```
[the remark this answers is not in the document]
Comment (mustafasrepo) [reply datafusion/physical-plan/src/aggregates/mod.rs:1033]: Yes you are right. I think the safest way is to check `num_rows == 1`. Changed accordingly

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Thanks @mustafasrepo and @ozankabak !

**Pull request merged**